ZipException when switching from sbt 1.0 to sbt 0.13

## steps
1.Take an existing project, port it to sbt 1.0
2. compile it with sbt 1.0
3. switch to a branch still using sbt 0.13, launch sbt

## problem
```scala

% sbt
[info] Loading global plugins from /home/smarter/.sbt/0.13/plugins
[info] Updating {file:/home/smarter/.sbt/0.13/plugins/}global-plugins...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
[info] Loading project definition from /home/smarter/opt/dotty/project
[info] Updating {file:/home/smarter/opt/dotty/project/}dotty-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] Done updating.
Error wrapping InputStream in GZIPInputStream: java.util.zip.ZipException: Not in GZIP format
        at sbt.ErrorHandling$.translate(ErrorHandling.scala:10)
        at sbt.WrapUsing.open(Using.scala:34)
        at sbt.Using.apply(Using.scala:23)
        at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:863)
        at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:862)
        at sbt.Using.apply(Using.scala:24)
        at sbt.IO$.gzipFileIn(IO.scala:862)
        at sbt.Sync$.readUncaught(Sync.scala:88)
        at sbt.Sync$.readInfo(Sync.scala:84)
        at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:28)
        at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:22)
        at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:948)
        at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:944)
        at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
        at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.util.zip.ZipException: Not in GZIP format
        at java.util.zip.GZIPInputStream.readHeader(GZIPInputStream.java:165)
        at java.util.zip.GZIPInputStream.<init>(GZIPInputStream.java:79)
        at sbt.Using$$anonfun$gzipInputStream$1.apply(Using.scala:84)
        at sbt.Using$$anonfun$gzipInputStream$1.apply(Using.scala:84)
        at sbt.Using$$anon$1.openImpl(Using.scala:51)
        at sbt.WrapUsing$$anonfun$open$2.apply(Using.scala:34)
        at sbt.ErrorHandling$.translate(ErrorHandling.scala:10)
        at sbt.WrapUsing.open(Using.scala:34)
        at sbt.Using.apply(Using.scala:23)
        at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:863)
        at sbt.IO$$anonfun$gzipFileIn$1.apply(IO.scala:862)
        at sbt.Using.apply(Using.scala:24)
        at sbt.IO$.gzipFileIn(IO.scala:862)
        at sbt.Sync$.readUncaught(Sync.scala:88)
        at sbt.Sync$.readInfo(Sync.scala:84)
        at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:28)
        at sbt.Sync$$anonfun$apply$1.apply(Sync.scala:22)
        at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:948)
        at sbt.Defaults$$anonfun$copyResourcesTask$1.apply(Defaults.scala:944)
        at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
        at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
[error] (compile:copyResources) Error wrapping InputStream in GZIPInputStream: java.util.zip.ZipException: Not in GZIP format
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore?
```

## expectation

sbt 0.13 should not crash, instead it should ignore the files it doesn't understand and overwrite them.

## notes

Currently the problem can be worked around by doing:
```shell
rm -rf **/target
```

--- a/build.sbt
+++ b/build.sbt
@@ -7,7 +7,7 @@ import Sxr.sxr
 // but can be shared across the multi projects.
 def buildLevelSettings: Seq[Setting[_]] = inThisBuild(Seq(
   organization := "org.scala-sbt",
-  version := "0.13.15-SNAPSHOT",
+  version := "0.13.16-SNAPSHOT",
   bintrayOrganization := Some(if (publishStatus.value == "releases") "typesafe" else "sbt"),
   bintrayRepository := s"ivy-${publishStatus.value}",
   bintrayPackage := "sbt",

--- a/main/actions/src/main/scala/sbt/Sync.scala
+++ b/main/actions/src/main/scala/sbt/Sync.scala
@@ -68,6 +68,7 @@ object Sync {
   }
 
   import java.io.{ File, IOException }
+  import java.util.zip.ZipException
   import sbinary._
   import Operations.{ read, write }
   import DefaultProtocol.{ FileFormat => _, _ }
@@ -82,7 +83,15 @@ object Sync {
 
   def readInfo[F <: FileInfo](file: File)(implicit infoFormat: Format[F]): RelationInfo[F] =
     try { readUncaught(file)(infoFormat) }
-    catch { case e: IOException => (Relation.empty, Map.empty) }
+    catch {
+      case e: IOException  => (Relation.empty, Map.empty)
+      case e: ZipException => (Relation.empty, Map.empty)
+      case e: TranslatedException =>
+        e.getCause match {
+          case e: ZipException => (Relation.empty, Map.empty)
+          case _               => throw e
+        }
+    }
 
   def readUncaught[F <: FileInfo](file: File)(implicit infoFormat: Format[F]): RelationInfo[F] =
     IO.gzipFileIn(file) { in =>

